crates/solver/src/settlement_access_list.rs

```diff
@@ -120,18 +188,12 @@ impl AccessListEstimating for TenderlyApi {
     async fn estimate_access_lists(
         &self,
         txs: &[TransactionBuilder<DynTransport>],
-    ) -> Vec<Result<AccessList>> {
-        futures::future::join_all(txs.iter().map(|tx| async {
-            let input = tx.data.clone().context("transaction data does not exist")?;
-            let from = tx
-                .from
-                .clone()
-                .context("transaction from does not exist")?
-                .address();
-            let to = tx.to.context("transaction to does not exist")?;
+    ) -> Result<Vec<Result<AccessList>>> {
+        Ok(futures::future::join_all(txs.iter().map(|tx| async {
+            let (from, to, input) = resolve_call_request(tx)?;
```

Comment:
In the context of price estimators @vkgnosis had some [reservations](https://github.com/gnosis/gp-v2-services/commit/d0fe1beda68c694065f0237ab2802f816832f9fc) about using `join_all()` to implement a similar function for 3rd party services which don't offer a batch request endpoint. I don't have a strong opinion here I just thought it's something to consider if you haven't already.

Reply:
For now I will assume it won't be an issue because we have a pro account. But in general, also not sure about this one.

Reply:
It is probably fine here because this is a more controlled environment. For the price estimator we were calling it different places with different number of queries so I didn't want an unconditional join_all.  The access list code is only used in one or two places.
What you might still want to do is introduce a parallelization limit by converting the input iterator to a stream and then doing https://docs.rs/futures/latest/futures/stream/trait.StreamExt.html#method.for_each_concurrent . Not required for the PR just an idea.